File: public/docs/_examples/lifecycle-hooks/ts/app/logger.service.ts
import {Injectable} from 'angular2/core';

@Injectable()
export class LoggerService {
  logs:string[] = [];
  prevMsg = '';
  prevMsgCount = 1;

  log(msg:string)  {
    if (msg === this.prevMsg) {
      // Repeat message; update last log entry with count.
      this.logs[this.logs.length-1] = msg + ` (${this.prevMsgCount+=1}x)`;
    } else {
      // New message; log it.
      this.prevMsg = msg;
      this.prevMsgCount = 1;
      this.logs.push(msg);
    }
  }

  clear()   {this.logs.length = 0;}

  // schedules a view refresh to ensure display catches up
  tick() {
    setTimeout(() => {
    // console.log('tick')
    }, 0);
  }
}

File: public/docs/_examples/lifecycle-hooks/dart/lib/logger_service.dart
import 'dart:async';

import 'package:angular2/core.dart';

@Injectable()
class LoggerService {
  List<String> logs = [];
  String _prevMsg = '';
  int _prevMsgCount = 1;

  void log(String msg) {
    if (msg == _prevMsg) {
      // Repeat message; update last log entry with count.
      logs[logs.length - 1] = "$msg (${_prevMsgCount += 1}x)";
    } else {
      // New message; log it.
      _prevMsg = msg;
      _prevMsgCount = 1;
      logs.add(msg);
    }
  }

  void clear() => logs.clear();

  // schedules a view refresh to ensure display catches up
  tick() => new Future(() {});
}
